Turbulence.ModelGenerator/Converter.cs

```diff
@@ -0,0 +1,502 @@
+using System.Globalization;
+using System.Runtime.Serialization;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Xml;
+
+namespace Turbulence.ModelGenerator;
+
+public static class Converter
+{
+    // Amount of unmatched types
+    private static int _unmatchedCount = 0;
+    
+    // Mapping from record name to the namespace it belongs to
+    private static readonly Dictionary<string, string> Namespaces = new()
+    {
+        { "MessageComponent", "DiscordMessageComponents" },
+    };
+
+    /// <summary>
+    /// Convert all markdown tables in a directory to C# records.
+    /// </summary>
+    /// <param name="tablesPath">The directory with JSON tables.</param>
+    /// <param name="modelPath">The directory to put the generated records.</param>
+    public static async Task Convert(Uri tablesPath, Uri modelPath)
+    {
+        Directory.CreateDirectory(modelPath.AbsolutePath);
+
+        // First, extract all record names and their namespaces
+        foreach (string tableDir in Directory.GetDirectories(tablesPath.AbsolutePath))
+        {
+            foreach (string tableFile in Directory.GetFiles(tableDir))
+            {
+                string nameSpace = tableDir.Split(Path.DirectorySeparatorChar)[^1];
+                string recordName = tableFile.Split(Path.DirectorySeparatorChar)[^1];
+
+                if (Namespaces.ContainsKey(recordName))
+                {
+                    // Check if existing namespace matches
+                    if (Namespaces[recordName] == nameSpace) continue;
+
+                    throw new Exception($"Record {recordName} is part of multiple namespaces: {Namespaces[recordName]} and {nameSpace}");
+                }
+                
+                Namespaces.Add(recordName, nameSpace);
+            }
+        }
+
+        // Assumes that the depth level is exactly 1 deep
+        foreach (string tableDir in Directory.GetDirectories(tablesPath.AbsolutePath))
+        {
+            string modelDir = Path.Combine(modelPath.AbsolutePath, tableDir.Split(Path.DirectorySeparatorChar)[^1]);
+            Directory.CreateDirectory(modelDir);
+            
+            foreach (string tableFile in Directory.GetFiles(tableDir))
+            {
+                string modelFile = Path.ChangeExtension(
+                    Path.Combine(modelDir, tableFile.Split(Path.DirectorySeparatorChar)[^1]), ".cs");
+
+                // Write generated .cs record to file
+                await File.WriteAllTextAsync(
+                    modelFile, await TableToRecord(tableFile));
+            }
+        }
+
+        if (_unmatchedCount > 0) Console.WriteLine($"\nTotal unmatched types: {_unmatchedCount}");
```

Comment:
Having a count is nice because it allows you to quickly see if you removed any of the umatched cases, especially if you have like a 100.
That combined with the fact that its almost instant anyways, and performance doesn't really matter in the first place, I think we should just keep this.